test/unit/category.test.ts

```diff
@@ -0,0 +1,108 @@
+import { mock, instance, when, verify, reset } from 'ts-mockito';
+import { Repository } from 'typeorm';
+import Category from '../../app/orm/entities/Category';
+import { Dependencies } from '../../app/Types';
+import CategoryModel from '../../app/models/CategoryModel';
+
+describe('Category Model unit tests', () => {
+  const categoryRepoMock = <Repository<Category>>(<unknown>mock(Repository));
+  const categoryRepoMockInstance = instance(categoryRepoMock);
+  const categoryModel = new CategoryModel(<Dependencies>{
+    categoryRepo: categoryRepoMockInstance
+  });
+  const category = new Category();
+  category.id = 1;
+  category.description = 'This is a cool category';
+  category.name = 'The Cool Category';
+  const categories = [category];
+  const emptyCategories: Category[] = [];
+
+  beforeEach(async () => {
+    reset(categoryRepoMock);
+  });
+
+  test('Get all categories', async () => {
+    expect.assertions(1);
+    when(categoryRepoMock.find()).thenResolve(categories);
+    await expect(categoryModel.getAllCategories()).resolves.toEqual(categories);
+    verify(categoryRepoMock.find()).called();
+  });
+
+  test('Get all categories, error test', async () => {
+    expect.assertions(1);
+    when(categoryRepoMock.find()).thenResolve(emptyCategories);
+    try {
+      await categoryModel.getAllCategories();
+    } catch (error) {
+      expect('' + error + '').toEqual('Error: Categories not found');
```

Comment:
Like in the success cases you use helper `resolves` use `rejects` helper in error cases. Also correct way of checking the boom error are like this:

![image](https://user-images.githubusercontent.com/12229968/62627673-87d69080-b932-11e9-9eae-d3fbab27ff98.png)